api/v1beta2/kustomization_types.go

```diff
@@ -153,6 +153,11 @@ type KustomizationSpec struct {
 	// +kubebuilder:validation:Enum=none;client;server
 	// +optional
 	Validation string `json:"validation,omitempty"`
+
+	// Components specifies relative paths to specifications of other Components
+	// via relative paths, absolute paths, or URLs.
```

Comment:
Absolute paths and URLs shouldn't be allowed here

Reply:
I have removed it from the description, but nothing prevents the user from doing it anyways (assuming `--no-remote-bases=false`).

Reply:
We need to block this, so far all the Kustomize features exposed in a Flux Kustomization have this restriction, e.g. you can't refer to external patches they must be inlined.

Reply:
I don't see how this is easily doable. According to [kustomize's remote targets documentation](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/remoteBuild.md) a remote target can look like:
- `https://github.com/kubernetes-sigs/kustomize//examples/multibases/dev/?ref=v1.0.6`
- `git@github.com:owner/repo`
- `ssh://git@github.com/owner/repo`

Kustomize has [a parser](https://github.com/kubernetes-sigs/kustomize/blob/84bd402cc0662c5df3f109c4f80c22611243c5f9/api/internal/git/repospec.go#L81-L100) for this, but I can't use it, as it is a `internal` package.

Reply:
After a bit more digging I found the [host parser logic](https://github.com/kubernetes-sigs/kustomize/blob/84bd402cc0662c5df3f109c4f80c22611243c5f9/api/internal/git/repospec.go#L231-L239) and copied it with `file://` removed.

Looks good? :)

Reply:
Yeap, thanks!